drivers/watchdog/bflb_wdt.c

```diff
@@ -20,31 +21,32 @@
 #define BFLB_REG_BASE 0x2000A500
 
 #define BFLB_REG_WFAR 0x9C
-#define BFLB_VAL_WFAR 0xBABA
+#define BFLB_WFAR_MAGIC 0xBABA
 
 #define BFLB_REG_WSAR 0xA0
-#define BFLB_VAL_WSAR 0xEB10
+#define BFLB_WSAR_MAGIC 0xEB10
 
 #define BFLB_REG_WVR 0x6C
 #define BFLB_REG_WCR 0x98
-#define BFLB_VAL_WCR 0x1
+#define BFLB_WCR_RESET_COUNTER BIT(0)
 
 #define BFLB_REG_WMER 0x64
-#define BFLB_VAL_WE 0x1
-#define BFLB_VAL_WRIE 0x2
+#define BFLB_WMER_WATCHDOG_ENABLE BIT(0)
+#define BFLB_WMER_RESET_SOURCE BIT(1)
 
 #define BFLB_REG_TCCR 0x00
-#define BFLB_MASK_CS_WDT 0xF00
-#define BFLB_VAL_CS_1K 0x200
+#define BFLB_MASK_CS_WDT GENMASK(11,8)
+#define BFLB_TCCR_CS_1K (0x2 << 8)
 
 #define BFLB_REG_TCDR 0xbc
-#define BFLB_MASK_TCDR 0xFF000000
-#define BFLB_VAL_TCDR 0xFF000000
+#define BFLB_MASK_TCDR GENMASK(31,24)
 
-#define BFLB_REG_WMR 0x68
-#define BFLB_MASK_WMR 0xffff
-#define BFLB_VAL_WMR 240
+// Max out CLKDIV at 255 ticks
+// This will end up with a tick every 250ms with 1K clock
+#define BFLB_TCDR_CLKDIV  (0xFF << 24)
```

Comment:
This value should be computed from the selected clock frequency and BFLB_TICK_DIV so it's only specified in one place.  Also I'm not sure if a granularity <1s is required, since the kernel watchdog framework is all based on multiples of seconds.

Reply:
I'll change the calculation, but this is the slowest the clock can run. We've got the slowest internal clock with the largest clock divider. The practical impact is that we can only have a max timeout of 2^14 instead of 2^16 which is still an excessively large 4.5 hours.

Reply:
Fixed with [9b9c8fd](https://github.com/arm000/linux-bl808/pull/8/commits/9b9c8fd8166217cf560fe3bc4509fd10aeadba0b)